drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -191,80 +172,108 @@ static void notify_event(const struct device *dev, struct gps_event *evt)
 	}
 }
 
-static int open_socket(struct gps_drv_data *drv_data)
+static void cancel_works(struct gps_drv_data *drv_data)
 {
-	drv_data->socket = nrf_socket(NRF_AF_LOCAL, NRF_SOCK_DGRAM,
-				      NRF_PROTO_GNSS);
+	(void)k_work_cancel_delayable(&drv_data->timeout_work);
+	(void)k_work_cancel_delayable(&drv_data->bump_priority_work);
+	(void)k_work_cancel_delayable(&drv_data->pvt_ntf_work);
+	(void)k_work_cancel_delayable(&drv_data->sat_stats_work);
+	(void)k_work_cancel_delayable(&drv_data->blocked_ntf_work);
```

Comment:
I'm not certain should these be handled a bit differently - zephyrproject-rtos/zephyr#33104